src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -666,8 +722,13 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                     // To handle that properly resolve() would have to support
                     // something like [`ambi_fn`](<SomeStruct as SomeTrait>::ambi_fn)
                     .or_else(|| {
-                        let item =
-                            resolve_associated_trait_item(did, module_id, item_name, ns, self.cx);
+                        let item = resolve_associated_trait_item(
+                            tcx.type_of(did),
```

Comment:
Is this a reliable call? (e.g. can we be sure that it doesn't panic for the possible input values here)

Reply:
should be fine